api/types/image/manifest.go

```diff
@@ -0,0 +1,99 @@
+package image
+
+import (
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type ManifestKind string
+
+const (
+	ManifestKindImage       ManifestKind = "image"
+	ManifestKindAttestation ManifestKind = "attestation"
+	ManifestKindUnknown     ManifestKind = "unknown"
+)
+
+type ManifestSummary struct {
+	// ID is the content-addressable ID of an image and is the same as the
+	// digest of the image manifest.
+	//
+	// Required: true
+	ID string `json:"ID"`
+
+	// Descriptor is the OCI descriptor of the image.
+	//
+	// Required: true
+	Descriptor ocispec.Descriptor `json:"Descriptor"`
+
+	// Indicates whether all the child content (image config, layers) is
+	// fully available locally
+	//
+	// Required: true
+	Available bool `json:"Available"`
+
+	// Size is the size information of the content related to this manifest.
+	// Note: These sizes only take the locally available content into account.
+	//
+	// Required: true
+	Size struct {
+		// Content is the size (in bytes) of all the locally present
+		// content in the content store (e.g. image config, layers)
+		// referenced by this manifest and its children.
+		// This only includes blobs in the content store.
+		Content int64 `json:"Content"`
+
+		// Total is the total size (in bytes) of all the locally present
+		// data (both distributable and non-distributable) that's related to
+		// this manifest and its children.
+		// This equal to the sum of [Content] size AND all the sizes in the
+		// [Size] struct present in the Kind-specific data struct.
+		// For example, for an image kind (Kind == ManifestKindImage),
+		// this would include the size of the image content and unpacked
+		// image snapshots ([Size.Content] + [ImageData.Size.Unpacked]).
+		Total int64 `json:"Total"`
```

Comment:
Wondering if these should be nilable in cases where we fail to determine the size? So we can differentiate between `0` and 🤷🏻 .

Reply:
Oh, that's a good one; I know we used `-1` for other situations, but those were `-1` means "we didn't check".

Using `omitempty` also has its own downsides of course (w.r.t. other tools trying to use the JSON as-is) 🤔

Reply:
Would we have cases where "one of these" failed? Or would they more likely "both" fail?

(slightly considering if omitting all of `Size` would work for that), but .. tricky).

We could also consider adding a `Warnings[]` array somewhere for non-fatal, but still useful failures

Reply:
Yeah I'd be against making it `omitempty` actually - 
`-1` is a bit meh to me, as it's really easy to get unnoticed, especially if you only add/subtract it.
Not sure how other languages could handle it, but in pure JSON I think it's fine to just use explicit `null` instead of having us use the `omitempty`?

Reply:
Yeah the `Warnings` would be nice too IMO.

Reply:
LOL, we'll just make it a string and put `"NaN"` in it. Problem solved